pwa-devdocs/src/tutorials/pwa-studio-fundamentals/manage-state/index.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Manage component and application state
+---
+
+Your components can respond to dynamic events in your application by keeping track of internal or global state.
+This tutorial provides the steps for creating a button that manages its own internal state and updating it to use data from the global application state.
+
+By the end of this tutorial, you will know how to add internal state to a component and use Peregrine's context hooks to access global state data.
+
+For more information on state management, read the [State management][] topic.
+
+{: .bs-callout .bs-callout-info}
+This tutorial requires you to have a project set up using the steps provided in the [Project Setup][] tutorial.
+
+## Create a new component
+
+For this part of the tutorial, you will create a React functional component called _MyComponent_.
+This component will render a button that toggles and displays an internal boolean state.
+
+### Create the component directory
+
+Create a new `MyComponent` directory under your project's `src/components` directory.
+
+```sh
+mkdir -p src/components/MyComponent
+```
+
+### Create the component
+
+Under `src/components/MyComponent`, create a `myComponent.js` file with the following content:
+
+```jsx
+import React, { useState, useCallback } from 'react';
+
+import Button from "@magento/venia-ui/lib/components/Button";
+
+const MyComponent = () => {
+    const [booleanStatus, setBooleanStatus] = useState(false);
+
+    const toggleStatus = useCallback(() => {
+        setBooleanStatus(!booleanStatus);
+    }, [booleanStatus]);
+
+    const text = booleanStatus ? 'On' : 'Off';
+
+    return <Button onClick={toggleStatus}>{text}</Button>;
+};
+
+export default MyComponent;
+```
+
+This component uses the [`useState()` React hook][] to keep track of an internal boolean state value.
+It defines a `toggleStatus()` function to toggle this boolean state, which is called when the user clicks the rendered Button component.
+When the status value changes, the text on the button also changes.
+
+### Export component from the index file
+
+Create a `src/components/MyComponent/index.js` file with the following content to set the default component export for the `MyComponent` directory.
+
+```js
+export {default} from './myComponent'
+```
+
+## Add component to the storefront
+
+Add this button to your application's Header component using steps similar to the steps described in the [Modify the site footer][] tutorial.
+
+## Access global state data
```

Comment:
This is fine for how we do things today but heads up we're really trying to get away from using these global context hooks.

Reply:
Sure thing. Once we implement an alternative, I can update this tutorial.